--- before/purescript/Test_Church.go
+++ integrated/purescript/Test_Church.go
@@ -69,6 +69,17 @@
 })
 	})
 	return cache_Test_Church_mulC__2514794121
+}
+
+var cache_Test_Church_fromInt__gopurs_counted_function_0 gopurs_runtime.Value
+var once_Test_Church_fromInt__gopurs_counted_function_0 sync.Once
+func Get_Test_Church_fromInt__gopurs_counted_function_0() gopurs_runtime.Value {
+	once_Test_Church_fromInt__gopurs_counted_function_0.Do(func() {
+		cache_Test_Church_fromInt__gopurs_counted_function_0 = gopurs_runtime.Func3(func(remaining_0_box gopurs_runtime.Value, callback_1_box gopurs_runtime.Value, result_2_box gopurs_runtime.Value) gopurs_runtime.Value {
+return gopurs_runtime.Int(Call_Test_Church_fromInt__gopurs_counted_function_0(remaining_0_box.IntVal, callback_1_box, result_2_box.IntVal))
+})
+	})
+	return cache_Test_Church_fromInt__gopurs_counted_function_0
 }
 
 var cache_Test_Church_fromInt gopurs_runtime.Value
@@ -232,12 +243,55 @@
 }
 }
 
+func Call_Test_Church_fromInt__gopurs_counted_function_0(remaining_0_loop int64, callback_1_loop gopurs_runtime.Value, result_2_loop int64) int64 {
+fromInt__gopurs_counted_function_0:
+for {
+if false { continue fromInt__gopurs_counted_function_0 }
+var remaining_0 int64 = remaining_0_loop
+_ = remaining_0
+var callback_1 gopurs_runtime.Value = callback_1_loop
+_ = callback_1
+var result_2 int64 = result_2_loop
+_ = result_2
+var __t0 int64
+{
+if (remaining_0) == (int64(0)) {
+__t0 = result_2
+goto end_branch_0
+} else {
+
+}
+}
+{
+remaining_0_loop = (remaining_0) - (int64(1))
+callback_1_loop = callback_1
+result_2_loop = gopurs_runtime.Apply(callback_1, gopurs_runtime.Int(result_2)).IntVal
+continue fromInt__gopurs_counted_function_0
+__t0 = func() int64 { panic("unreachable") }()
+}
+end_branch_0:
+return __t0
+}
+}
+
 func Call_Test_Church_fromInt(v_0_loop int64) gopurs_runtime.Value {
 fromInt:
 for {
 if false { continue fromInt }
 var v_0 int64 = v_0_loop
 _ = v_0
+var __t2 gopurs_runtime.Value
+{
+if (v_0) >= (int64(0)) {
+__t2 = gopurs_runtime.Func2(func(f_2 gopurs_runtime.Value, x_3 gopurs_runtime.Value) gopurs_runtime.Value {
+return gopurs_runtime.Int(Call_Test_Church_fromInt__gopurs_counted_function_0(v_0, f_2, x_3.IntVal))
+})
+goto end_branch_2
+} else {
+
+}
+}
+{
 var __t1 gopurs_runtime.Value
 {
 if (v_0) == (int64(0)) {
@@ -256,7 +310,10 @@
 })
 }
 end_branch_1:
-return __t1
+__t2 = __t1
+}
+end_branch_2:
+return __t2
 }
 }
 
