--- ../gopurs/src/Gopurs/CodeGen.purs
+++ ../gopurs/src/Gopurs/CodeGen.purs
@@ -1077,7 +1077,8 @@
             expectedGoTypeForVal = exprTypeToGoType (unsafePerformEffect (Ref.read helpersRef)).pointerAdtPaths modNameStr (getExprType val)
             
             newBound = Map.insert ident { name: ident, goType: expectedGoTypeForVal } bound
-            resBody = translateExprImpl_ helpersRef (depth + 1) modNameStr recVars moduleArities newBound Nothing loopCtx isTail false resVal.nextId body
+            typedBody = if expectedTy /= Any then setTcoExprType expectedTy body else body
+            resBody = translateExprImpl_ helpersRef (depth + 1) modNameStr recVars moduleArities newBound Nothing loopCtx isTail false resVal.nextId typedBody
           in
             { stmts: resVal.stmts <> StmtLeaf (GoRaw ("var " <> ident <> " " <> goTypeToStr expectedGoTypeForVal <> " = " <> printGoExpr (coerceGoExpr resVal.expr resVal.exprType expectedGoTypeForVal))) <> StmtLeaf (GoRaw ("_ = " <> ident)) <> resBody.stmts, expr: resBody.expr, exprType: resBody.exprType, nextId: resBody.nextId }
 
