--- gopurs/src/Gopurs/CodeGen.purs
+++ gopurs/src/Gopurs/CodeGen.purs
@@ -534,14 +534,6 @@
-    case expr of
-      Typed type_ a ->
-        let
-          aWithAnn = setTcoExprAnnType type_ a
-          res = translateExprImpl_ helpersRef depth modNameStr recVars moduleArities bound tcoIdent loopCtx isTail inEffectBlock nextId aWithAnn
-          expectedGoType = exprTypeToGoType (unsafePerformEffect (Ref.read helpersRef)).pointerAdtPaths modNameStr type_
-        in
-          { stmts: res.stmts, expr: coerceGoExpr res.expr res.exprType expectedGoType, exprType: expectedGoType, nextId: res.nextId }
