--- before/Test.LazyEvaluation.fs
+++ after/Test.LazyEvaluation.fs
@@ -3,6 +3,8 @@
 
 open System
 open System.Collections.Generic
+
+let rec private Test_LazyEvaluation_buildThunks_thunk_native (sharpurs_thunk_local_0: int) (sharpurs_thunk_local_1: (unit -> int)) : (unit -> int) = (if (sharpurs_thunk_local_0 = (0)) then sharpurs_thunk_local_1 else (Test_LazyEvaluation_buildThunks_thunk_native ((sharpurs_thunk_local_0 - (1))) ((fun (sharpurs_thunk_local_2: unit) -> ((sharpurs_thunk_local_1 ()) + (1))))))
 
 let Test_LazyEvaluation_Lazy  = (box (fun (x: obj) -> (box x)))
 
@@ -16,7 +18,7 @@
 and Test_LazyEvaluation_buildThunks = box (fun (v: obj) ->  (fun (v1: obj) -> Test_LazyEvaluation_buildThunks_tco v v1))
 
 
-let rec Test_LazyEvaluation_runManyTimes_tco (v: obj) (v1: obj) : obj = ((match (((unbox ((box v))), (unbox ((box v1))))) with | (LitInt 0 (), acc) -> ((box acc)) | (n, acc) -> ((Test_LazyEvaluation_runManyTimes_tco ((box ((unbox<int> (box ((box n)))) - (unbox<int> (box ((box 1))))))) ((box ((unbox<int> (box ((box acc)))) + (unbox<int> (box ((sharpurs_apply (box ((box Test_LazyEvaluation_force))) (box ((sharpurs_apply (box ((sharpurs_apply (box ((box Test_LazyEvaluation_buildThunks))) (box ((box 1000)))))) (box ((sharpurs_apply (box ((box Test_LazyEvaluation_defer))) (box ((box (fun (v2: obj) -> (box 0))))))))))))))))))))))
+let rec Test_LazyEvaluation_runManyTimes_tco (v: obj) (v1: obj) : obj = ((match (((unbox ((box v))), (unbox ((box v1))))) with | (LitInt 0 (), acc) -> ((box acc)) | (n, acc) -> ((Test_LazyEvaluation_runManyTimes_tco ((box ((unbox<int> (box ((box n)))) - (unbox<int> (box ((box 1))))))) ((box ((unbox<int> (box ((box acc)))) + (unbox<int> (box ((box ((Test_LazyEvaluation_buildThunks_thunk_native ((1000)) ((fun () -> (0)))) ()))))))))))))
 and Test_LazyEvaluation_runManyTimes = box (fun (v: obj) ->  (fun (v1: obj) -> Test_LazyEvaluation_runManyTimes_tco v v1))
 
 
